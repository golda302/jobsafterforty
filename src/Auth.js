import React, { Component } from "react";
import { connect } from "react-redux";
import { openAuth, logoutUser } from "./actions/auth";
import C from "./constants/auth.js";

const Auth = props => {

  switch (props.auth.status) {

    case C.AUTH_LOGGED_IN:
      return (
        <p>
          <span>Logged in as {props.auth.username}.</span>
          {" "}<button onClick={props.logoutUser}>Log out</button>
        </p>
      );
      break;
    case C.AUTH_AWAITING_RESPONSE:
      return (
        <p>
          <button disabled>authenticating...</button>
        </p>
      );
      break;
    default:
      return (
        <p>
          <button onClick={props.openAuth}>Log in</button>
        </p>
      );
  }
};

const mapStateToProps = state => ({ auth: state.auth });

const mapDispatchToProps = {
  openAuth,
  logoutUser
};

export default connect(mapStateToProps, mapDispatchToProps)(Auth);
