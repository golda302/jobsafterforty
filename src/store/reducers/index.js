import { combineReducers } from 'redux';
import jobs from './jobs';
import searchJobs from './jobs';
import auth from './auth';
import feedback from './feedback';
import leftmenu from './leftmenu';
import regions from './regions';
import scopes from './scopes';
import jobfields from './jobfields';
//import SearchRegion from './SearchRegion';

const rootReducer = combineReducers({
  jobs,
  searchJobs,
  auth,
  feedback,
  leftmenu,
  regions,
  scopes,
  jobfields,
});


export default rootReducer;
