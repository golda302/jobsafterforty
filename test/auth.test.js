import React from 'react';
import Auth from '../src/Auth';
import renderer from 'react-test-renderer';
import { render } from 'enzyme';
import { Provider } from "react-redux";
import store from "../src/store/index.js";
import C from "../src/constants/auth.js";
test('If logged in, shows username', () => {
	var auth = {status: C.AUTH_LOGGED_IN, username: "anat"};
  	const component = renderer.create(
  	<Provider store={store}>
  			<Auth auth={auth} logoutUser="logoutUser()"></Auth>
  			</Provider>);
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
test('When logged out, shows: "log in"', () => {
	var auth = {status: C.AUTH_ANONYMOUS};
  	const component = renderer.create(
  	<Provider store={store}>
  			<Auth auth={auth} logoutUser="logoutUser()"></Auth>
  			</Provider>);
  var tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})