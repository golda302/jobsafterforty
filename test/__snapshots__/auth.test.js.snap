exports[`test If logged in, shows username 1`] = `
<p>
  <button
    onClick={[Function]}>
    Log in
  </button>
</p>
`;

exports[`test When logged out, shows: "log in" 1`] = `
<p>
  <button
    onClick={[Function]}>
    Log in
  </button>
</p>
`;
